src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/CountReadsSpark.java

```diff
@@ -0,0 +1,53 @@
+package org.broadinstitute.hellbender.tools.spark.pipelines;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.SparkProgramGroup;
+import org.broadinstitute.hellbender.engine.spark.SparkCommandLineProgram;
+import org.broadinstitute.hellbender.engine.spark.datasources.ReadsSparkSource;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.io.File;
+import java.io.IOException;
+
+@CommandLineProgramProperties(summary = "Counts reads in the input BAM", oneLineSummary = "Counts reads in a BAM file", programGroup = SparkProgramGroup.class)
+public final class CountReadsSpark extends SparkCommandLineProgram {
+
+    private static final long serialVersionUID = 1L;
+
+    @Argument(doc = "uri for the input bam, either a local file path, hdfs:// path, or a gs:// bucket path",
+            shortName = StandardArgumentDefinitions.INPUT_SHORT_NAME, fullName = StandardArgumentDefinitions.INPUT_LONG_NAME,
+            optional = false)
+    public String bam;
+
+    @Argument(doc = "uri for the output file: a local file path",
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            optional = true)
+    public String out;
+
+    @Override
+    protected void runPipeline(final JavaSparkContext ctx) {
+        final ReadsSparkSource readSource = new ReadsSparkSource(ctx);
+        final JavaRDD<GATKRead> reads = readSource.getParallelReads(bam, null);
+        final long count = reads.count();
+        System.out.println(count);
+        if (out != null){
+            final File file = new File(out);
+            try {
+                FileUtils.write(file, String.valueOf(count));
```

Comment:
no, the doc was confusing. it accepts a null if i'm not writing to cloud storage